### PR TITLE
Use pkg-config to link to xnvme

### PR DESCRIPTION
**Pull request by Joelgranados**
Signed-off-by: Joel Granados <j.granados@samsung.com>

### CODE REVIEW

**Comment by Joelgranados** [comment]
@jwdevantier @nmtadam. This is no longer WIP as xNVMe is v28 is out. I suggest we push this after the daemon goes in as it applies cleanly on top.

**Review by nmtadam — CHANGES_REQUESTED** [review]
Can we squash this down to one commit? This also does not cleanly apply onto master. I tried just adding the changes but the tests aren not correctly running for me. I got a fla_daemon_open as an undefined symbol.

**Comment by Joelgranados** [comment]
> squashing down and committing. @nmtadam 

Of course, I was just planning to do it when we do the merge. So select squash and merge instead of rebase and merge.

**Comment by Joelgranados** [comment]
@nmtadam : just did the squash when I did the rebase

**Comment by nmtadam** [comment]
This breaks the python bindings for me.

```
tests/test_libflexalloc.py FFFFF                                                                                                                                                                   [ 76%]
tests/test_mkfs.py FF                                                                                                                                                                              [ 84%]
tests/test_pool_root.py FF                                                                                                                                                                         [ 92%]
tests/test_sync.py FF 
```

**Comment by Joelgranados** [comment]
@nmtadam : I do not see the failure. are you sure you are in `e8a1897`.
It might be that you are in a previous version of the PR.

File: meson.build [hunk 1]
```diff
@@ -13,13 +13,7 @@ if get_option('FLEXALLOC_XNVME_IGNORE_MDTS')
...
 
 ### Dependencies ###
-thread_dep = dependency('threads')
-uuid_dep = dependency('uuid')
-numa_dep = cc.find_library('libnuma', dirs:['/usr/lib/x86_64-linux-gnu'], required : true)
-aio_dep = cc.find_library('libaio', dirs:['/usr/lib/x86_64-linux-gnu'], required : true)
-rt_dep = cc.find_library('rt', required: true)
-xnvme_dep = cc.find_library('libxnvme', dirs:['/usr/lib/', '/usr/local/lib'], required : true)
-xnvme_deps = [thread_dep, uuid_dep, numa_dep, aio_dep, rt_dep, xnvme_dep]
+xnvme_deps = dependency('libxnvme', version : '>=0.0.27' )
```
[the remark this answers is not in the document]
Comment (jwdevantier) [reply meson.build]: I think for now it's OK to move fast along-side xNVMe, but Adam has a valid point. Down the road, we should figure out if we can avoid requiring everyone to use the very, very latest xNVMe (and especially kernel, if that applies, as it's a mess to upgrade on production systems).

**Pull request merged**